presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSource.java

```diff
@@ -361,6 +364,15 @@ public ConnectorPageSource getPageSource()
         if (toType instanceof VarcharType && (fromHiveType.equals(HIVE_BYTE) || fromHiveType.equals(HIVE_SHORT) || fromHiveType.equals(HIVE_INT) || fromHiveType.equals(HIVE_LONG))) {
             return new IntegerNumberToVarcharCoercer<>(fromType, (VarcharType) toType);
         }
+        if (toType instanceof VarcharType && fromHiveType.equals(HIVE_FLOAT)) {
+            return new FloatToVarcharCoercer((VarcharType) toType);
+        }
+        if (toType instanceof VarcharType && fromHiveType.equals(HIVE_DOUBLE)) {
+            return new DoubleToVarcharCoercer((VarcharType) toType);
+        }
+        if (toType instanceof VarcharType && (fromType instanceof DecimalType)) {
```

Comment:
Why this one is expressed in terms of `fromType`, while others use `fromHiveType`?

Reply:
HiveType's `equals` implementation assumes types are equal if type name is equal which is not correct/safe for parametrised types like Varchar or Decimal.

Here we need to take this branch if the `fromHiveType` is some sort of Decimal type. This is handled similar to how other Decimal types are handled further down at https://github.com/prestosql/presto/pull/4426/files/761af9d991d7bdd2fae99bfd63f7833aa372056e#diff-17a9f6c64f9dd9a24bf1ae969e6c7257R394.